### PR TITLE
Embed interop type for PackageReference

### PR DESCRIPTION
**Pull request by jainaashish**
## Bug
Fixes: https://github.com/NuGet/Home/issues/2365
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: This PR is the implementation of supporting interop type assemblies with PackageReference. More details about the solution can be found in the [Spec doc](https://github.com/NuGet/Home/wiki/Embed-Interop-Types-with-PackageReference)

## Testing/Validation
Tests Added: Working on adding tests
Reason for not adding tests:  
Validation done:

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs [hunk 1]
```diff
@@ -526,6 +531,17 @@ internal ManagedCodePatterns(ManagedCodeConventions conventions)
...
+
+                EmbedAssemblies = new PatternSet(
+                    conventions.Properties,
+                    groupPatterns: new PatternDefinition[]
+                        {
+                            new PatternDefinition("embed/{tfm}/{any?}", table: DotnetAnyTable),
+                        },
+                    pathPatterns: new PatternDefinition[]
+                        {
+                            new PatternDefinition("embed/{tfm}/{assembly}", table: DotnetAnyTable),
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs:543]: We really should revisit the meaning of 'any' in our pattern definitions :) 

It's an inconsistent behavior right now, this is a note to myself, more than something relevant for this PR. :)

File: test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs [hunk 2]
```diff
@@ -581,6 +581,36 @@ public void PackTaskLogic_SupportsContentFiles_WithPackageFlatten()
             }
         }
 
+        [Fact]
+        public void PackTaskLogic_EmbedInteropAssembly()
```
Comment (nkolev92) [inline test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs:585]: Can you add a DotnetIntegrationTest as well? 

The targets => PackRequest conversion is not covered and that one could easily be changed by someone in the future and the test wouldn't catch it :)

**Review by nkolev92 — APPROVED** [review]
LGTM. 

Please make sure to add a DotnetIntegrationTest for the pack scenario.

**Comment by AArnott** [comment]
The PR description promises tests will come later, and describes no validation done. Was any done?

Now that this has been in for a long while, I figured it was safe to assume most customers have the update and I could actually update my packages to use this pattern. But it seems to not work (#10188). I wonder if I'm doing something wrong -- I sure hope so, since I'd hate to have to reset the clock and wait another two years before a fix for this feature is widespread so I can actually use this feature.

**Pull request merged**